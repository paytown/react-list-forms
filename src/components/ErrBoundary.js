import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class ErrBoundary extends Component {
  static propTypes = {
    children: PropTypes.node
  }

  state = {
    error: false
  }

  static getDerivedStateFromError() {
    return { error: true };
  }

  componentDidCatch(error) {
    console.error(error);
  }

  render() {
    if(this.state.error) {
      return <h3>Err!</h3>;
    }

    return this.props.children;
  }
}
