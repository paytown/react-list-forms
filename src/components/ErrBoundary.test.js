import React from 'react';
import { shallow } from 'enzyme';
import ErrBoundary from './ErrBoundary';

describe('err boundary', () => {
  it('throws err instead of breaking page', () => {
    const wrapper = shallow(<ErrBoundary />);
    expect(wrapper).toMatchSnapshot();
  });
});
