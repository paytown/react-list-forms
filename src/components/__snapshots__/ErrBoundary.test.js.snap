// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`err boundary throws err instead of breaking page 1`] = `""`;
